pkg/deployer/updater/updater.go

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ */
+
+package updater
+
+import (
+	"fmt"
+	"github.com/k8stopologyawareschedwg/deployer/pkg/deployer/platform"
+	"github.com/k8stopologyawareschedwg/deployer/pkg/tlog"
+)
+
+type Type string
+
+const (
+	RTE Type = "RTE"
+	NFD Type = "NFD"
+)
+
+type Options struct {
+	Platform          platform.Platform
+	WaitCompletion    bool
+	UpdaterConfigData string
+	PullIfNotPresent  bool
+	UpdaterType       string
+}
+
+func Deploy(log tlog.Logger, opts Options) error {
```

Comment:
or just create two subpackages and expose the functions directly. Simpler, less code.

Reply:
The intention was to keep the API unchanged and hide the implementation details, but ok I'll change it.

Reply:
in general it's a good thing to care about, but in this case noone is consuming our packages, `deployer` is alpha-level and it is a big change (version-bump worthy) so let's take the simpler approach.
We didn't make any compatibility promise yet, nor we will in the foreseeable future.